ConsoleApp1/Program.cs

```diff
@@ -1,9 +1,478 @@
+                    searchArray[int.Parse(indexOrder)].firstName = Console.ReadLine() ?? throw new Exception(); ;
+
+                    editContact = TransformArray(searchArray);
+                    File.WriteAllLines(pathPhoneBook, editContact);
+
+                    break;
+
+                case ConsoleKey.D3:
+
+                    Console.Clear();
+                    Console.WriteLine("Rewrite Phone Number:\t");
+
+                    searchArray[int.Parse(indexOrder)].phoneNuber = Console.ReadLine() ?? throw new Exception(); ;
+
+                    editContact = TransformArray(searchArray);
+
+                    try
+                    {
+                        File.WriteAllLines(pathPhoneBook, editContact);
+                    }
+                    catch (Exception)
+                    {
+                        throw;
+                    }
+
+                    break;
+                default:
+                    break;
+            }
+
+        }
+        //
+        static void DeleteContact()
+        {
+            (string number, string lastName, string firstName, string phoneNuber)[] searchArray = GetStringArrayFromFile(pathPhoneBook);
+            var DelArray = new (string number, string lastName, string firstName, string phoneNuber)[searchArray.Length - 1];
+            string[] deleteContact = new string[searchArray.Length];
+            string paramsSearch;
+
+            Console.WriteLine("Enter (first name or last name or number) of the contact you want to delete");
+            paramsSearch = Console.ReadLine() ?? throw new Exception(); ;
+            Search(paramsSearch);
+
+            Console.Write("Select the order in Result:\t");
+            string indexOrder;
+
+            try
+            {
+                indexOrder = Console.ReadLine();
+            }
+            catch (FormatException)
+            {
+                throw;
+            }
+
+            try
+            {
+                Console.WriteLine(searchArray[int.Parse(indexOrder)]);
+            }
+            catch (FormatException)
+            {
+                throw;
+            }
+
+            for (int i = 0; i < searchArray.Length && i < int.Parse(indexOrder); i++)
+                DelArray[i] = searchArray[i];
+            for (int i = int.Parse(indexOrder) + 1; i < searchArray.Length; i++)
+            {
+                DelArray[i - 1] = searchArray[i];
+                DelArray[i - 1].number = $"{i - 1}";
+            }
+            deleteContact = TransformArray(DelArray);
+            File.WriteAllLines(pathPhoneBook, deleteContact);
+        }
+        //
+        static void SortAscDesc()
+        {
+            var searchArray = GetStringArrayFromFile(pathPhoneBook);
+            Console.WriteLine("Select sort:\n\t1. Asc\n\t2. Desc");
+            int sort;
+
+            try
+            {
+                sort = int.Parse(Console.ReadLine());
+            }
+            catch (Exception)
```

Comment:
Try to specify exactly the types of exceptions and handle them (for instance, prompt user to input again), Exception is a general type